mesonbuild/ast/printer.py

```diff
@@ -12,6 +12,37 @@
 import typing as T
 
 
+# Also known as "order of operations" or "binding power".
+# This is the counterpart to Parser.e1, Parser.e2, Parser.e3, Parser.e4, Parser.e5, Parser.e6, Parser.e7, Parser.e8, Parser.e9
+def precedence_level(node: mparser.BaseNode) -> int:
```

Comment:
Wouldn't be better to add the precedence level as an attribute of the Node objects? Also, why do you use 10, 20, 30 here, then change them to 1, 2, 3 in the following commit?

Reply:
I think having the precedence levels as an attribute of the Node objects would be harder to read. My way, we have one function where you can see which operater takes precedence over which other operator.

You're right about the other thing tho, I squashed the changes into one commit.